core/src/rpc.rs

```diff
@@ -52,6 +52,14 @@ pub struct JsonRpcConfig {
     pub faucet_addr: Option<SocketAddr>,
 }
 
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RpcSignatureStatusConfig {
+    #[serde(flatten)]
+    pub commitment: Option<CommitmentConfig>,
```

Comment:
I'm a little bothered by how the API allows me to say: `{"searchTransactionHistory":true, "commitment":"recent"}`.  "I want recent commitment but also search the transaction history" -- I'm not sure what that means.  

`"searchTransactionHistory":true` only returns rooted transactions from blockstore, even when `"commitment":"recent"`.    So if we're not making roots for more than a couple minutes, it looks like `{"searchTransactionHistory":true, "commitment":"recent"}` could first return a valid result, followed by an empty result once that signature is flushed from the status cache, and then finally a valid result again once the next root is finally made.  

I think it makes sense for `searchTransactionHistory` and `commitment` to be mutually exclusive but that makes the API feel weird too.   Perhaps we split this into two RPC APIs after all?  `getSignatureStatuses` remains unchanged, and we add a `getConfirmedSignatureStatuses` that requires no parameters and behaves like `{"searchTransactionHistory":true, "commitment":"max"}`

Reply:
@mvines The status cache actually doesn't remove any forks while the node is not making roots (https://github.com/solana-labs/solana/blob/7b68628e6cfdda47adbcb14a1dc3087151cda778/runtime/src/status_cache.rs#L161), so your example of an empty result can't happen. (It is of course up for discussion whether this is how the status_cache *should* be working)

To me, `{"searchTransactionHistory":true, "commitment":"recent"}` means "I want this transaction at any commitment level in any slot; I don't necessarily know if it's recent enough for commitment to be relevant, and I don't mind spending the resources to look in the history if it isn't", which doesn't seem all that crazy.

I'm not super excited about that 2nd RPC api.

Reply:
Ever since we added `confirmations` to the status response, I feel it doesn't sense to have a `commitment` param. Can we drop `commitment` for this endpoint?

Reply:
I'm okay with dropping `commitment`. Would having only one config param alleviate any of your concerns, @mvines ?

Reply:
Oh nice, yep that sounds great

Reply:
✔️